### DISCUSSION TITLE
What do you think about 0.3 and what do you wish for 0.4?

### DISCUSSION THREAD

**Discussion by Avnsx** [Avnsx/fansly#10] [Q&A]
Do you like the updates, or is there something bothering you? Let me know!

**Comment by Redworthy** [comment]
Is there any way for images to be downloaded in the same order as they appear in posts?

**Reply by Avnsx** [reply]
no; just downloads in the order that the api responds with

**Comment by MugWumpster** [comment]
Sorry, new to Github and this may be the wrong spot to comment. I used your program and it worked beautifully. Just wanted to thank you! Best wishes.

**Comment by coxtique99** [comment]
Hello, I am a noob aswell wanted to thank you I enjoy your work. It will be very nice if it could download just certain collections from a creator. Like just "videos". I don't know if there is any way

**Reply by Avnsx** [reply]
Collections support is not added yet (https://github.com/Avnsx/fansly/issues/12) and I personally will never add it. This issue is entirely up to contributors.

**Comment by freiheitner** [comment]
If some videos from my fansly creator are not downloaded, even after setting update_recent_download = True, is there any other way to pull them?  Out of 138 files I got 133, including the most recent, but not 2 released prior to that.

Also, the hashing of my already downloaded content completely locked my system (could not even move the mouse) until it completed.  This is on a Ryzen 5800X with 16GB of RAM.  Any way to make hashing either lower priority or use only X number of threads?

**Reply by Avnsx** [reply]
``Out of 138 files I got 133, including the most recent, but not 2 released prior to that.``
Did you make sure that you've actual access to the images through your account? If you do and it still did not download them, did you check the console if it said anything about not supported file extensions? Else just name the creator and so I can check myself and debug the code

``completely locked my system (could not even move the mouse) until it completed``
I'm not sure how this happend, I wrote this code using a AMD Ryzen 5 5600x - 6 core processor. I've left the threads for the hashing function at default arguments on purpose.
https://github.com/Avnsx/fansly/blob/57e90741b1b4a120b85791424bf0386ba5c4d14e/fansly_scraper.py#L185


The [documentation for ThreadPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ThreadPoolExecutor) says:
*Changed in version 3.8: Default value of max_workers is changed to min(32, os.cpu_count() + 4). This default value preserves at least 5 workers for I/O bound tasks. It utilizes at most 32 CPU cores for CPU bound tasks which release the GIL. And it avoids using very large resources implicitly on many-core machines.*

Which was the reason there's no manual specification for threading, since this implemented algorithm should always find the optimal count of threads to avoid system stutters. I guess you had your system under a CPU heavy task at the time? Or maybe you've applied bad overclocking settings, incase you overclocked I would recommend using programs like https://github.com/sp00n/corecycler to make sure that your CPU cores are not running into errors and that way providing worse performance then they should.

I don't really plan on adding a custom setting into the configuration file for the amount of threads, as most of the people will not understand what number to put there and it'll perform worse then it would normally. I mean you could also use the raw python source and just add the argument ``max_workers = mycustomnumber`` here for both ThreadPoolExecutor definitions in line 185 (marked above) and line 193 and see how you'll end up then.

``Any way to make hashing either lower priority or use only X number of threads?``
You could [set the python processes affinity lower](https://winbuzzer.com/2020/07/03/how-to-set-cpu-affinity-in-windows-10-with-task-manager-or-system-properties-xcxwbt/) in task manager, that would limit the amount of useable cores. In addition you could lower the processes priority there too.

Also on a side note I think you did not really undersatnd what ``update_recent_download = True`` does; here's a longer explanation: https://github.com/Avnsx/fansly/discussions/16

**Reply by freiheitner** [reply]
I should have said, I am running the .exe version not the python source.  I have Python installed so I can try switching over to that.  Probably also because I'm using the .exe I don't have (or don't know where to find) the console to look for "unsupported file type" messages.  Yes, I can play the videos which did not download just fine when logged into fansly in my browser.

The freeze happened twice--I restarted my computer after the first time.  The only other thing I had running was a single Youtube video in Firefox.  On the 2nd run I only had fansly scraper running.  Now that I (feel that) I know why my system froze, it's not a major issue--I won't be trying to download multiple times per day or anything.

I had read up on update_recent_download prior to downloading the app.  I only ran with it because those 5 files did not download and I was hoping it might pick them up on a second run.  It's running right now and indeed it looks like it is picking them up, so yay!

Really liking the app.  Glad to have found it, and I have starred this project.

**Comment by NumNuts101** [comment]
> Do you like the updates, or is there something bothering you? Let me know!

Would there be any way to add more than one creator at a time?

**Reply by Avnsx** [reply]
Yes but I don't see a point in adding that, you can just download one by one and you'll end up with the same result.

**Comment by ZilockeT** [comment]
Probably totally off-topic, but any chance of adding more sites? I would absolutely love a program like yours for OnlyFans, for example.

Also, thanks a lot for creating this! It works great.

**Comment by Anonymouse1746** [comment]
Fantastic repo, love the minimalist style. I bet it cuts down on customer service quite a lot? Starred without hesitation!

Please consider adding a field in the config.ini for us to input our local adjustment to epoch time, and have the filenames based off of that. For example, US EST would be -5, Central Europe CET would be +1, etc. It would be less jarring to see the files match our timezones, especially if the content creator is also in said timezone.

I totally understand if you don't want to do that, to keep things simple and cut down on confused users. If that's the case, it would be great if you specified in the wiki or FAQ or both that 'Datepost' uses UTC aka GMT aka Epoch time. I figured it out when I found issue #28, but otherwise it doesn't appear to be documented. Also the [documentation](https://github.com/Avnsx/fansly/wiki/Explanation-of-provided-programs-&-their-functionality#4-configini) still lists that option as 'Date_posted' even though it has been changed to 'Datepost'.

Thank you very much!
